### PR TITLE
Support Layout Animations on macOS

### PR DESCRIPTION
**Pull request by jfedak**
## Summary

Previously, Layout Animations were disabled on macOS platform (see #4912). This PR adds this functionality.

In order to make Layout Animations work on macOS, several changes had to be made:

- add `center` property to RCTUIView - our initial idea was to just add this property to REAUIView, but this approach was not sufficient;
- change the way of getting `absolutePosition` - according to Apple documentation (see [Appkit docs](https://developer.apple.com/documentation/appkit/nsview/1483406-convertpoint?language=objc) and [UIKit docs](https://developer.apple.com/documentation/uikit/uiview/1622442-convertpoint?language=objc)), there is no need to pass specific view to `convertPoint` method if we want to get window based coordinates. This is a correct approach on macOS, since we can't pass NSWindow to this method;
- use different method to get window's size on macOS - instead of `bounds.size` we used `frame.size`;
- set `header = nil` on macOS - native stack is not available on macOS, thus we are sure the first view is not a header

## Test plan

Tested on macOS (working fine) and iOS (not breaking anything)

### CODE REVIEW

File: apple/LayoutReanimation/REASnapshot.m [hunk 1]
```diff
@@ -44,15 +47,19 @@ - (void)makeSnapshotForView:(REAUIView *)view useAbsolutePositionOnly:(BOOL)useA
     _values[@"originXByParent"] = [NSNumber numberWithDouble:view.center.x - view.bounds.size.width / 2.0];
     _values[@"originYByParent"] = [NSNumber numberWithDouble:view.center.y - view.bounds.size.height / 2.0];
 
+#if TARGET_OS_OSX
+    REAUIView *header = nil;
```
[the remark this answers is not in the document]
Comment (jfedak) [reply apple/LayoutReanimation/REASnapshot.m:51]: Native stack is not available on macOS, thus there is no such thing as header in this case.

**Review by piaskowyk — APPROVED** [review]

**Review by tomekzaw — APPROVED** [review]
:shipit:

**Comment by Saadnajmi** [comment]
Hey, just came across this. I see that you're extending RCTUIView to add stuff like `center`, would it be helpful if that change was made upstream in React Native macOS so that you don't have to maintain the extension here?

**Comment by jfedak** [comment]
Hi @Saadnajmi! It would be awesome to see it in macos repo. Is there any reason why you decided not to implement `center` prop in RCTUIView before?

**Comment by Saadnajmi** [comment]
> Hi @Saadnajmi! It would be awesome to see it in macos repo. Is there any reason why you decided not to implement `center` prop in RCTUIView before?

We didn't need it, is the simplest. The iOS react native code never referred to it so we didn't need to shim it basically.

**Comment by Saadnajmi** [comment]
Feel free to make a PR if you like, otherwise I'll try and get to it.

**Comment by jfedak** [comment]
> Feel free to make a PR if you like, otherwise I'll try and get to it.

Sure, gonna make it in a day or two 😄

**Pull request merged**